### ISSUE TITLE
Stateless component support works how?

### ISSUE THREAD

**Issue by matsad111** [Pomax/react-onclickoutside#175]
It looks like this library works great and the functionality would be perfect for my app, and while the docs say stateless components are supported, how is not exactly clear.  I have a redux app using stateless functional components with the following attempts at using this library:
Attempt 1: (I didn't really expect this to work because the function isn't exposed)
``` 
const SomeComponent = ({ ...props, hideComponent }) => {
  const handleClickOutside = e => hideComponent() 
  return (
     //jsx goes here - the hideComponent function is already used here
  )
}

export default onClickOutside(SomeComponent) 
```
Attempt 2: 

```
import store from '../store/'

const SomeComponent = ({...props }) => {  //props include hideComponent function
  return ( 
    //jsx
  )
}

export default onClickOutside(SomeComponent, e => store.dispatch(hideComponent()) )
```
Attempt 3: 
```
import store from '../store/'

const SomeComponent = ({...props }) => {  //props include hideComponent function
  return ( 
    //jsx
  )
}

const handleClickOutside = e => {
  e.preventDefault()
  store.dispatch(hideComponent())
}

export default onClickOutside(SomeComponent, handleClickOutside )
```
Is it possible to use this functionality without class components?

**Comment by Pomax** [comment]
If you're using a function, not a class, pass the handler as config option to the HOC
```
const SomeComponent = props => {
  return (
     // jsx goes here
  )
}

export default onClickOutside(SomeComponent, {
  handleClickOutside: function(instance) {
    // do something here
  }
});
```

**Comment by matsad111** [comment]
Thanks for the reply, but I still can't get it to work.  I tried several variations on what you suggested, but I'm still getting this error:
```
Uncaught Error: ReactOnRails encountered an error while rendering component: App.
Original message: Component lacks a function for processing outside click events specified by the handleClickOutside config option.
```
Everything I've tried is some variation on `store.dispatch(actionCreator()))` or just `console.log('something')`, it doesn't require some specific sort of function or keyword, does it?

**Comment by Pomax** [comment]
Rather than variations, can you please show the *actual* code you tried? Because the code in https://github.com/Pomax/react-onclickoutside/issues/175#issuecomment-299582536 supplies a `handleClickOutside` and should not trigger any errors. If it does, that would be a bug.

**Comment by matsad111** [comment]
Sure, here you go:
```
import React from 'react'
import PropTypes from 'prop-types'
import onClickOutside from 'react-onclickoutside'

import store from '../store/store'
import { showMenu } from '../actions/'

const AutoExtendMenu = ({ shown,
                          hiddenStyles, shownStyles,
                          className,
                          children,
                          showMenu,
                          hideMenu}) => {

  const styleClass = shown ? shownStyles : hiddenStyles

  const onClick = () => {
    if (shown === true) {
      hideMenu()
    }
    else {
      showMenu()
    }
  }

  return (
      <div className={className}>
        <div className={styleClass.panel}>
          {children}
          <div onClick={onClick} className={styleClass.button} >
            <span className={styleClass.upButton}></span>
          </div>
        </div>
        <div onClick={onClick} className={styleClass.button} >
          <span className={styleClass.downButton}></span>
        </div>

      </div>
  )
}

AutoExtendMenu.propTypes = {
  shown: PropTypes.bool,
  hiddenStyles: PropTypes.object,
  shownStyles: PropTypes.object,
  className: PropTypes.string,
  children: PropTypes.node,
  hideMenu: PropTypes.func.isRequired,
  showMenu: PropTypes.func.isRequired,
}

export default onClickOutside(AutoExtendMenu, {
  handleClickOutside: e => {store.dispatch(showMenu('hide top'))},
})
```

**Comment by matsad111** [comment]
I also tried: 
```
//all else the same

const handleClickOutside = e => {
  store.dispatch(showMenu('hide top'))
}

export default onClickOutside(AutoExtendMenu, {
  handleClickOutside,},
})
```

**Comment by matsad111** [comment]
And 
```
export default onClickOutside(AutoExtendMenu, {
  handleClickOutside: function(e){ console.log('closing the menu')} },
})
```

**Comment by Pomax** [comment]
Would it be possible for you to reduce that code to a form that does not rely on a mystery "store"? That way it can be turned into a running example and we can do some debugging on it.

**Comment by delasky** [comment]
the way the its currently written, the handleClickOutside property in the config object should be an function that takes the instance and returns another function that takes the event. 

export default onClickOutside(AutoExtendMenu, {
  handleClickOutside: function(instance){ return function(e) {console.log('closing the menu')} },
})

should work

https://github.com/Pomax/react-onclickoutside/blob/master/index.js#L138

not sure if this is the intended behavior or not.

Thanks @Pomax for the super useful library

**Comment by matsad111** [comment]
@Pomax I wasn't going to be able to handle closing the menu without invoking the `store` as the app is built in Redux with stateless functional components and that's how I'm handling all the state changes with Redux action creator functions.  I was, however, able to make it work thanks to @delasky 's suggestion.  Here's what I've got right now: 
```
export default onClickOutside(AutoExtendMenu, {
  handleClickOutside: () => {
    return () => {
      console.log('closing the menu');
      store.dispatch(showMenu('hide top'))
    }
  }
})
```
I tested a variety of permutations from what @delasky suggested to what I have written here and this is the least verbose version, and most in keeping with how a lot of Redux stuff I've seen is written.  I won't be using the library where I thought I would for some reasons that have to do with other  issues, but I'll definitely keep it mind for next time.  Thanks very much, guys!
